lib/metasploit/framework/password_crackers/cracker.rb

```diff
@@ -0,0 +1,452 @@
+                  }, if: 'fork.present?'
+
+        validates :hash_path, :'Metasploit::Framework::File_path' => true, if: 'hash_path.present?'
+
+        validates :pot, :'Metasploit::Framework::File_path' => true, if: 'pot.present?'
+
+        validates :max_runtime,
+                  numericality: {
+                      only_integer:             true,
+                      greater_than_or_equal_to: 0
+                  }, if: 'max_runtime.present?'
+
+        validates :max_length,
+                  numericality: {
+                      only_integer:             true,
+                      greater_than_or_equal_to: 0
+                  }, if: 'max_length.present?'
+
+        validates :wordlist, :'Metasploit::Framework::File_path' => true, if: 'wordlist.present?'
+
+        # @param attributes [Hash{Symbol => String,nil}]
+        def initialize(attributes={})
+          attributes.each do |attribute, value|
+            public_send("#{attribute}=", value)
+          end
+        end
+
+        # This method takes a {framework.db.cred.private.jtr_format} (string), and
+        # returns the string number associated to the hashcat format
+        #
+        # @param[String] a jtr_format string
+        # @return [String] the format number for Hashcat
+        def jtr_format_to_hashcat_format(format)
+          case format
+          when 'md5crypt'
+            return '500'
+          when 'descrypt'
+            return '1500'
+          when 'bsdicrypt'
+            return '12400'
+          when 'sha256crypt'
+            return '7400'
+          when 'sha512crypt'
+            return '1800'
+          when 'bcrypt'
+            return '3200'
+          when 'lm', 'lanman'
+            return '3000'
+          when 'nt', 'ntlm'
+            return '1000'
+          when 'mssql'
+            return '131'
+          when 'mssql05'
+            return '132'
+          when 'mssql12'
+            return '1731'
+          # hashcat requires a format we dont have all the data for
+          # in the current dumper, so this is disabled in module and lib
+          #when 'oracle', 'des,oracle'
+          #  return '3100'
+          when 'oracle11', 'raw-sha1,oracle'
+            return '112'
+          when 'oracle12c', 'pbkdf2,oracle12c'
+            return '12300'
+          when 'postgres', 'dynamic_1034', 'raw-md5,postgres'
+            return '12'
+          when 'mysql'
+            return '200'
+          when 'mysql-sha1'
+            return '300'
+          when 'PBKDF2-HMAC-SHA512' #osx 10.8+
+            return '7100'
+          when 'xsha' #osx 10.4-6
+            return '122'
+          when 'xsha512' #osx 10.7
+            return '1722'
+          when 'PBKDF2-HMAC-SHA1' #Atlassian
+            return '12001'
+          when 'phpass' #Wordpress/PHPass, Joomla, phpBB3
+            return '400'
+          when 'mediawiki' # mediawiki b type
+            return '3711'
+          end
+          nil
```

Comment:
Seeing as this method does nothing outside of the `case` block, you could move this `nil` into an `else` and remove every instance of `return`, as the value returned from the `case` will be returned by the method, as the `case` is the effectively last line in the method. ie:

```ruby
        def jtr_format_to_hashcat_format(format)
          case format
          when 'md5crypt'
            '500'
          when '...'
            '...'
          else
            nil
          end
        end
```

```
irb(main):001:0>         def jtr_format_to_hashcat_format(format)
irb(main):002:1>           case format
irb(main):003:2>           when 'md5crypt'
irb(main):004:2>             '500'
irb(main):005:2>           when '...'
irb(main):006:2>             '...'
irb(main):007:2>           else
irb(main):008:2>             nil
irb(main):009:2>           end
irb(main):010:1>         end
=> :jtr_format_to_hashcat_format
irb(main):011:0> jtr_format_to_hashcat_format('asdf')
=> nil
irb(main):012:0> jtr_format_to_hashcat_format('md5crypt')
=> "500"
irb(main):013:0> 
```

Reply:
I like this, my python brain didn't think in this way, but it does save a lot of `return` statements for the same result!